### PR TITLE
feat(clang-tidy): add .clang-tidy-ignore

### PR DESCRIPTION
**Pull request by yhisaki**
## Description

In this PR, we've introduced a `.clang-tidy-ignore` file. By using this file and [this Action](https://github.com/autowarefoundation/autoware-github-actions/pull/326), we can exclude specific files from the clang-tidy check.

## Related links

- https://github.com/autowarefoundation/autoware-github-actions/pull/326

## How was this PR tested?

Tested in [this Action](https://github.com/autowarefoundation/autoware.universe/actions/runs/12063019141/job/33639712216).

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: .github/workflows/build-and-test-differential.yaml [hunk 1]
```diff
@@ -105,12 +105,12 @@ jobs:
 
       - name: Run clang-tidy
         if: ${{ steps.get-changed-files.outputs.changed-files != '' }}
-        uses: autowarefoundation/autoware-github-actions/clang-tidy@v1
+        uses: yhisaki/autoware-github-actions/clang-tidy@enhance-clang-tidy
```
Comment (xmfcx) [inline .github/workflows/build-and-test-differential.yaml:108]: - https://github.com/autowarefoundation/autoware-github-actions/pull/326 is merged.

Now we can make a release and point to it. @yhisaki do you need help with releasing?

File: common/autoware_time_utils/src/time_utils/time_utils.cpp [hunk 2]
```diff
@@ -17,6 +17,8 @@
 #include <limits>
 #include <stdexcept>
 
+// change
```
Comment (xmfcx) [inline common/autoware_time_utils/src/time_utils/time_utils.cpp:20]: Can you also make a change in a package from perception to see if ignore is working?

Comment (xmfcx) [reply common/autoware_time_utils/src/time_utils/time_utils.cpp:20]: Current `report.log`:

```log
Enabled checks:
    bugprone-dangling-handle

clang-apply-replacements version 14.0.0
clang-tidy-14 --use-color -export-fixes /tmp/tmpyffx98p7/tmpulc2pgnq.yaml -p=build/ /__w/autoware.universe/autoware.universe/common/autoware_time_utils/src/time_utils/time_utils.cpp
Writing fixes to /tmp/clang-tidy-result/fixes.yaml ...
```

File: .github/workflows/build-and-test-differential.yaml (hunk 1 above)
Comment (yhisaki) [reply .github/workflows/build-and-test-differential.yaml:108]: @xmfcx 
Thank you! I don't know much about a release and would like your help.

Comment (yhisaki) [reply .github/workflows/build-and-test-differential.yaml]: Thank you very much for the clear instruction!!!
I released [v1.45.0](https://github.com/autowarefoundation/autoware-github-actions/releases/tag/v1.45.0) 🎉

Comment (xmfcx) [reply .github/workflows/build-and-test-differential.yaml]: Perfect!

File: common/autoware_time_utils/src/time_utils/time_utils.cpp (hunk 2 above)
Comment (yhisaki) [reply common/autoware_time_utils/src/time_utils/time_utils.cpp:20]: Tested in [here](https://github.com/autowarefoundation/autoware.universe/actions/runs/12065875880/job/33645777711?pr=9498)!
Screenshot from 2024-11-28 18-27-24

File: common/autoware_time_utils/src/time_utils/time_utils.cpp [hunk 3]
```diff
@@ -17,6 +17,8 @@
...
+
```
Comment (xmfcx) [inline common/autoware_time_utils/src/time_utils/time_utils.cpp:21]: It works, thanks for your efforts!

We can remove the dummy changes :+1:

**Review by xmfcx — APPROVED** [review]

File: common/autoware_time_utils/src/time_utils/time_utils.cpp (hunk 3 above)
Comment (xmfcx) [reply common/autoware_time_utils/src/time_utils/time_utils.cpp]: done in https://github.com/autowarefoundation/autoware.universe/pull/9498/commits/267e43363177fad37f3854401cd75934ed48e26f

Comment (yhisaki) [reply common/autoware_time_utils/src/time_utils/time_utils.cpp]: Thank you!
I reverted the dummy changes!

**Pull request merged**